presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcMetadata.java

```diff
@@ -302,6 +304,29 @@ public JdbcTableHandle getTableHandle(ConnectorSession session, SchemaTableName
         return Optional.of(new LimitApplicationResult<>(handle, jdbcClient.isLimitGuaranteed(session)));
     }
 
+    @Override
+    public Optional<TableScanRedirectApplicationResult> applyTableScanRedirect(ConnectorSession session, ConnectorTableHandle table)
+    {
+        JdbcTableHandle tableHandle = (JdbcTableHandle) table;
+
+        List<JdbcColumnHandle> sourceColumns = tableHandle.getColumns()
+                .map(columnSet -> (List<JdbcColumnHandle>) columnSet.stream().map(JdbcColumnHandle.class::cast).collect(toImmutableList()))
+                .orElseGet(() -> jdbcClient.getColumns(session, tableHandle));
+
+        Optional<JdbcTableScanRedirectionResult> result = jdbcClient.getRedirectedTableScan(session, tableHandle, sourceColumns);
+        if (result.isEmpty()) {
+            return Optional.empty();
+        }
+
+        Map<ColumnHandle, String> columnsMapping = result.get().getDestinationColumns();
```

Comment:
we shouldn't output filter back here, because in the future (if we support redirection after aggregtation pushdown) filters might be on unaggregated columns (before aggregation happens)